### PR TITLE
 [foreman] [foreman_proxy] Add ALL_PROXY and NO_PROXY to collected env_vars

### PR DESCRIPTION
**Pull request by pafernanr**
Add ALL_PROXY and NO_PROXY to collected env_vars

Resolves: #3788

**Signed-off-by: Pablo Fernández Rodríguez <pafernan@redhat.com>**
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [ x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [ x] Is the subject and message clear and concise?
- [ x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x ] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?
- [x ] Are all passwords or private data gathered by this PR [obfuscated](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#how-to-prevent-collecting-passwords)?

### CODE REVIEW

File: sos/report/plugins/foreman.py [hunk 1]
```diff
@@ -297,7 +297,12 @@ def collect_proxies(self):
                                         timeout=10)
 
         # collect http[|s]_proxy env.variables
-        self.add_env_var(["http_proxy", "https_proxy"])
+        self.add_env_var([
+            'HTTP_PROXY',
+            'HTTPS_PROXY',
+            'NO_PROXY',
+            'ALL_PROXY'
```
Comment (jcastill) [inline sos/report/plugins/foreman.py:304]: little thing here, but can you add a comma at the end?

File: sos/report/plugins/foreman_proxy.py [hunk 2]
```diff
@@ -42,7 +42,12 @@ def setup(self):
         ])
 
         # collect http[|s]_proxy env.variables
-        self.add_env_var(["http_proxy", "https_proxy"])
+        self.add_env_var([
+            'HTTP_PROXY',
+            'HTTPS_PROXY',
+            'NO_PROXY',
+            'ALL_PROXY'
```
Comment (jcastill) [inline sos/report/plugins/foreman_proxy.py:49]: same here, comma at the end of 'ALL_PROXY'

**Review by jcastill — CHANGES_REQUESTED** [review]

**Comment by jcastill** [comment]
@pafernanr looks good to me, but can you squash them all together?

**Comment by pafernanr** [comment]
@jcastill thanks for your tips.

**Comment by arif-ali** [comment]
@pafernanr can you squash the commits, so that we only see one commit please. That's what @jcastill was suggesting

**Review by pmoravec — APPROVED** [review]
ACK to the change itself.

Just the commit contains also one commit from main :-o and manual merge must be done.

**Pull request closed without merging**